core/blockchain.go

```diff
@@ -1183,8 +1150,9 @@ func (bc *BlockChain) InsertReceiptChain(blockChain types.Blocks, receiptChain [
 	// this function only accepts canonical chain data. All side chain will be reverted
 	// eventually.
 	writeAncient := func(blockChain types.Blocks, receiptChain []types.Receipts) (int, error) {
-		var batch = bc.db.NewBatch()
 		first := blockChain[0]
+
+		// TODO: do we need to check this for all blocks?
```

Comment:
I don't see why that would be needed. We need to check that we only ever insert linked data into ancients. It's enough to check the first block, _as long as the `blockChain` being inserted here is linked internally.